javascripts/discourse/initializers/initialize-for-header-icon-links.gjs

```diff
@@ -1,22 +1,15 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
-import { iconNode } from "discourse-common/lib/icon-library";
+import icon from "discourse-common/helpers/d-icon";
 import { dasherize } from "@ember/string";
 import isValidUrl from "../lib/isValidUrl";
-import { h } from "virtual-dom";
 
-function buildIcon(icon) {
-  if (isValidUrl(icon)) {
-    return h(
-      "img",
-      {
-        attributes: {
-          src: icon,
-        },
-      },
-      ""
-    );
+function buildIcon(iconNameOrImageUrl) {
+  if (isValidUrl(iconNameOrImageUrl)) {
+    return <template>
+      <img src="{{iconNameOrImageUrl}}" aria-hidden="true"/>
+    </template>
   } else {
-    return iconNode(icon.toLowerCase());
+    return <template>{{icon iconNameOrImageUrl }}</template>
```

Comment:
```suggestion
    return <template>{{icon iconNameOrImageUrl}}</template>
```

There is an extra space here, is that intended?

Reply:
Nope! Thanks for the suggestions :)